src/components/QuranReader/TranslationView/BookmarkIcon.tsx

```diff
@@ -1,10 +1,10 @@
 import Button, { ButtonType } from 'src/components/dls/Button/Button';
-import { useSelector } from 'react-redux';
+import { shallowEqual, useSelector } from 'react-redux';
 import { selectBookmarks } from 'src/redux/slices/QuranReader/bookmarks';
 import StarIcon from '../../../../public/icons/star.svg';
 
 const BookmarkIcon = ({ verseKey }: { verseKey: string }) => {
-  const { bookmarkedVerses } = useSelector(selectBookmarks);
+  const { bookmarkedVerses } = useSelector(selectBookmarks, shallowEqual);
```

Comment:
Does `shallowEqual` work with objects? because `bookmarkedVerses` is an object

Reply:
yeah, I think we should refactor bookmarkedVerses and use areArrayEqual